x-pack/plugins/enterprise_search/common/types/indices.ts

```diff
@@ -15,6 +15,10 @@ import {
 import { Connector } from './connectors';
 import { Crawler } from './crawler';
 
+export interface AlwaysShowPattern {
+  alias_pattern?: 'search-';
```

Comment:
suggestion: We should probably type these as `string`. Your definition now makes it so the only allowed strings for this interface are `search-` and `.ent-search-engine-documents`. If that's intentional, you're probably better off making this a constant rather than an interface.

Reply:
Absolutely! We could make the `search-` and `.ent-search-engine-documents` as string constants. But, from my understanding, this may require adding a variable in the  getIndexData and fetchIndices. (As existing are string variables that points to `search-` )

With the interface, we could probably make the ` alias_pattern and index_pattern` (in interface AlwaysShowPattern) to an array,  which would help us adding additional pattern(s) if required in future.

Reply:
Ah, sorry, let me rephrase this. Right now you've defined this interface, which is a type definition, as: 
```
export interface AlwaysShowPattern {
  alias_pattern?: 'search-';
  index_pattern?: '.ent-search-engine-documents';
}
```
So this means that the *only* possible values those patterns can have in the function are:
```
export interface AlwaysShowPattern {
  alias_pattern: 'search-',
  index_pattern: '.ent-search-engine-documents',
}
```
They cannot hold other string values right now. Which means that using these values as an input doesn't really do anything right now: you're passing them around, but they're always the same when defined. 

Instead, if you define the interface as: 
```
export interface AlwaysShowPattern {
  alias_pattern?: string;
  index_pattern?: string;
}
```
the interface can also hold other string values.

I wouldn't worry too much about preparing for possible future changes--we want to refactor these functions anyway, as they're too complicated and trying to do too many different things at the same time. It's likely that we're not going to need that flexibility anyway--and if we do, it won't be hard to add later regardless of the solution chosen now.

Reply:
Thank you @sphilipse  for detail explanation. I agree. I will make the relevant changes.